internal/component/otelcol/exporter/splunkhec/splunkhec.go

```diff
@@ -0,0 +1,85 @@
+// package splunkhec provides an otel.exporter splunkhec component
+
+package splunkhec
+
+import (
+	"errors"
+
+	"github.com/grafana/alloy/internal/component"
+	"github.com/grafana/alloy/internal/component/otelcol"
+	otelcolCfg "github.com/grafana/alloy/internal/component/otelcol/config"
+	"github.com/grafana/alloy/internal/component/otelcol/exporter"
+	splunkhec_config "github.com/grafana/alloy/internal/component/otelcol/exporter/splunkhec/config"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/splunkhecexporter"
+	otelcomponent "go.opentelemetry.io/collector/component"
+	otelextension "go.opentelemetry.io/collector/extension"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:      "otelco.exporter.splunkhec",
+		Community: true,
+		Args:      Arguments{},
+		Exports:   otelcol.ConsumerExports{},
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			fact := splunkhecexporter.NewFactory()
+			return exporter.New(opts, fact, args.(Arguments), exporter.TypeAll)
+		},
+	})
+}
+
+type Arguments struct {
+	Client splunkhec_config.SplunkHecClientArguments `alloy:"client,block"`
+	Queue  otelcol.QueueArguments                    `alloy:"sending_queue,block,optional"`
+	Retry  otelcol.RetryArguments                    `alloy:"retry_on_failure,block,optional"`
+
+	// Splunk specific configuration settings
+	Splunk splunkhec_config.SplunkConf `alloy:"splunk,block"`
+
+	// DebugMetrics configures component internal metrics. Optional.
+	DebugMetrics otelcolCfg.DebugMetricsArguments `alloy:"debug_metrics,block,optional"`
+}
+
+var _ exporter.Arguments = Arguments{}
+
+func (args *Arguments) SetToDefault() {
+	args.Queue.SetToDefault()
+	args.Retry.SetToDefault()
+	args.Client.SetToDefault()
+	args.Splunk.SetToDefault()
+	args.DebugMetrics.SetToDefault()
+}
+
+func (args Arguments) Convert() (otelcomponent.Config, error) {
+
+	return splunkhec_config.SplunkHecArguments{
+		Splunk:                   args.Splunk,
+		QueueSettings:            *args.Queue.Convert(),
+		RetrySettings:            *args.Retry.Convert(),
+		SplunkHecClientArguments: args.Client,
+	}, nil
+
+}
+
+func (args *Arguments) Validate() error {
```

Comment:
in the validate we should have the same checks as in the otel config. I found out that the config is invalid if LogDataEnabled and ProfilingDataEnabled are both set to false. This should be documented in the .md file